tests/integration/autoTracking.spec.js

```diff
@@ -0,0 +1,770 @@
+          unstruct_event: {
+            data: {
+              schema: 'iglu:com.snowplowanalytics.snowplow/focus_form/jsonschema/1-0-0',
+              data: {
+                elementId: 'lname',
+              }
+            } 
+          }
+        },
+      })
+    ).toBe(true)
+  })
+
+  it('should send focus_form on included element', () => {
+    expect(
+      logContains({
+        event: {
+          event: 'unstruct',
+          app_id: 'autotracking',
+          page_url: 'http://snowplow-js-tracker.local:8080/form-tracking.html?filter=include',
+          unstruct_event: {
+            data: {
+              schema: 'iglu:com.snowplowanalytics.snowplow/focus_form/jsonschema/1-0-0',
+              data: {
+                elementId: 'lname',
+              }
+            } 
+          }
+        },
+      })
+    ).toBe(true)
+  })
+
+  it('should send focus_form on element included by filter', () => {
+    expect(
+      logContains({
+        event: {
+          event: 'unstruct',
+          app_id: 'autotracking',
+          page_url: 'http://snowplow-js-tracker.local:8080/form-tracking.html?filter=filter',
+          unstruct_event: {
+            data: {
+              schema: 'iglu:com.snowplowanalytics.snowplow/focus_form/jsonschema/1-0-0',
+              data: {
+                elementId: 'fname',
+              }
+            } 
+          }
+        },
+      })
+    ).toBe(true)
+
+    expect(
+      logContains({
+        event: {
+          event: 'unstruct',
+          app_id: 'autotracking',
+          page_url: 'http://snowplow-js-tracker.local:8080/form-tracking.html?filter=filter',
+          unstruct_event: {
+            data: {
+              schema: 'iglu:com.snowplowanalytics.snowplow/focus_form/jsonschema/1-0-0',
+              data: {
+                elementId: 'lname',
+              }
+            } 
+          }
+        },
+      })
+    ).toBe(false)
+  })
+
+  it('should not send focus_form on elements', () => {
+    expect(
+      logContains({
+        event: {
+          event: 'unstruct',
+          app_id: 'autotracking',
+          unstruct_event: {
+            data: {
+              schema: 'iglu:com.snowplowanalytics.snowplow/focus_form/jsonschema/1-0-0',
+              data: {
+                elementId: 'excluded-fname',
+              }
+            } 
+          }
+        },
+      })
+    ).toBe(false)
+  })
+
+  it('should not send submit_form', () => {
+    expect(
+      logContains({
+        event: {
+          event: 'unstruct',
+          app_id: 'autotracking',
+          unstruct_event: {
+            data: {
+              schema: 'iglu:com.snowplowanalytics.snowplow/submit_form/jsonschema/1-0-0',
+              data: {
+                formId: 'excludedForm',
+                formClasses: [ 'excluded-form' ]
+              }
+            } 
+          }
+        }
+      })
+    ).toBe(false)
+  })
+})
```

Comment:
Love this. Something else that strikes me is that anywhere we have multiple events on a page, maybe we should enable the web page context and check that it behaves correctly. (eg just a count of distinct values matching the amount of page loads we expect).

Might save us effort in thinking of a way to check this directly if we ever need to.